x-pack/legacy/plugins/apm/public/context/UrlParamsContext/index.tsx

```diff
@@ -12,19 +12,36 @@ import React, {
   useState
 } from 'react';
 import { withRouter } from 'react-router-dom';
-import { uniqueId } from 'lodash';
+import { uniqueId, mapValues, pick } from 'lodash';
 import { IUrlParams } from './types';
 import { getParsedDate } from './helpers';
 import { resolveUrlParams } from './resolveUrlParams';
 import { UIFilters } from '../../../typings/ui-filters';
+import {
+  localUIFilterNames,
+  LocalUIFilterName
+} from '../../../server/lib/ui_filters/local_ui_filters/config';
 
 interface TimeRange {
   rangeFrom: string;
   rangeTo: string;
 }
 
-function useUiFilters({ kuery, environment }: IUrlParams): UIFilters {
-  return useMemo(() => ({ kuery, environment }), [kuery, environment]);
+function useUiFilters(
+  params: Pick<IUrlParams, 'kuery' | 'environment' | LocalUIFilterName>
+): UIFilters {
+  return useMemo(() => {
+    const { kuery, environment, ...localUIFilters } = params;
+    const mappedLocalFilters = mapValues(
+      pick(localUIFilters, localUIFilterNames) as Record<
+        LocalUIFilterName,
+        IUrlParams[LocalUIFilterName]
+      >,
```

Comment:
It seems we are using `_.pick` quite a bit which causes us to manually type it. This is error-prone so I suggest using a type-safe version. There might be libs that do this, or we can roll our own:

```ts
function pick<T, K extends keyof T>(obj: T, ...keys: K[]) {
  return keys.reduce(
    (acc, key) => {
      acc[key] = obj[key];
      return acc;
    },
    {} as Pick<T, K>
  );
}
```

Reply:
I like it! I wonder if our types get a lot better if we upgrade to lodash 4.x.